packages/ra-core/src/dataProvider/useMutation.ts

```diff
@@ -95,9 +97,37 @@ const useMutation = (
         loaded: false,
     });
     const dataProvider = useDataProvider();
+    const dataProviderWithDeclarativeSideEffects = useDataProviderWithDeclarativeSideEffects();
+
     const mutate = useCallback(
         (event, callTimePayload = {}, callTimeOptions = {}): void => {
             setState({ loading: true });
+
+            if (options.withDeclarativeSideEffectsSupport) {
+                dataProviderWithDeclarativeSideEffects(
```

Comment:
instead of pasting the code, I'd create a variable function and call it once:

```js
const dataProviderWithSideEffects = options.withDeclarativeSideEffectsSupport ? dataProviderWithDeclarativeSideEffects : dataProvider;
```